stm32: clock control: fix on H7 + style edit

erwango opened the pull request:
soc: stm32h7: Move PWR init code in soc init function
    
  Move PWR init code out of clock control driver and
  put SMPS related function under SMPS condition as it
  is not supported by all soc variants of the series.
    
Fixes #22363

Additionally:
Rework CONFIG_CLOCK_STM32_HSE_BYPASS related code to make
this part of the code more readable.
    
Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
soc: stm32h7: Move PWR init code in soc init function

Move PWR init code out of clock control driver and
put SMPS related function under SMPS condition as it
is not supported by all soc variants of the series.

Fixes #22363

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers/clock_control: stm32: HSE_BYPASS code cleanup

Rework CONFIG_CLOCK_STM32_HSE_BYPASS related code to make
this part of the code more readable.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@mkainy, can you check this is fixing your issue?

mkainy commented on drivers/clock_control/clock_stm32_ll_common.c:
```diff
@@ -376,11 +378,11 @@ static int stm32_clock_control_init(struct device *dev)
-#else
-		LL_RCC_HSE_DisableBypass();
-#endif /* CONFIG_CLOCK_STM32_HSE_BYPASS */
+		if (IS_ENABLED(CONFIG_CLOCK_STM32_HSE_BYPASS) {
```
")" Is missing. It will cause a compile error.

erwango replied:
Thanks!

mkainy commented on soc/arm/st_stm32/stm32h7/soc_m7.c:
```diff
@@ -85,6 +85,15 @@ static int stm32h7_init(struct device *arg)
 	SystemCoreClock = 64000000;
 
+	/* Power Configuration */
+#if SMPS
```
The compile result is shown below.
soc_m7.c: 89: 9: error: #if with no expression #if SMPS

erwango committed:
board: stm32h474i_disco: Small fixes on m4 core

ST-Link VPC is wired to UART_1.
Console should be assigned to this peripheral to be available.
By default UART_1 is assigned to m7 core and not available to m4.

This commit does not change this, but makes it easier to change
UART_1 assignment from m7 to m4 core.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@mkainy Issues should be fixed now. Can you check this is working?

galak commented:
Will wait to see if @mkainy is able to test this out.

mkainy commented:
I have confirmed that the code is performing properly.
